geniza/corpus/management/commands/import_iiif_urls.py

```diff
@@ -0,0 +1,77 @@
+import csv
+import re
+import logging
+from collections import namedtuple
+
+from django.core.management.base import BaseCommand
+
+from geniza.corpus.models import Fragment
+
+
+# logging config: use levels as integers for verbosity option
+logger = logging.getLogger("import")
+logging.basicConfig()
+LOG_LEVELS = {0: logging.ERROR, 1: logging.WARNING, 2: logging.INFO, 3: logging.DEBUG}
+
+
+class Command(BaseCommand):
+    """Given a CSV of fragments and IIIF URLs, add those IIIF urls to their respective fragments in the database"""
+
+    def __init__(self, *args, **options):
+        self.csv_path = options.get("csv")
+        self.overwrite = options.get("overwrite")
+        self.dryrun = options.get("dryrun")
+
+    def add_arguments(self, parser):
+        parser.add_argument("-c", "--csv", type=str, required=True)
+        parser.add_argument("-o", "--overwrite", action="store_true")
+        parser.add_argument("-d", "--dryrun", action="store_true")
+
+    def handle(self, *args, **options):
+        self.csv_path = options.get("csv")
+        self.overwrite = options.get("overwrite")
+        self.dryrun = options.get("dryrun")
+
+        rows = self.get_iiif_csv()
+        for row in rows:
+            self.import_iiif_url(row)
+
+    def get_iiif_csv(self):
+        # given a name for a file in the configured data import urls,
+        # load the data by url and initialize and return a generator
+        # of namedtuple elements for each row
+
+        with open(self.csv_path) as f:
+            csvreader = csv.reader(f)
+            header = next(csvreader)
+
+            # Create a namedtuple based on headers in the csv
+            # and local mapping of csv names to access names
+            CsvRow = namedtuple("IiifCsvRow", header)
+
+            # iterate over csv rows and yield a generator of the namedtuple
+            for row in csvreader:
+                yield CsvRow(*row)
```

Comment:
would [`DictReader`](https://docs.python.org/3/library/csv.html#csv.DictReader) be simpler than creating a `namedtuple` type for rows? it should automatically parse the headers.

Reply:
fwiw, the namedtuple approach in the data import was a workaround so we could have nice names for the fields in  code without changing the headers in spreadsheet; I usually prefer dictreader